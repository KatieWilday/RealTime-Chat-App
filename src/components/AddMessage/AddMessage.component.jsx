import React from "react";
import { firestore } from "../../firebase/firebase.utils";

import { styleFunc, inputHelper } from "../../utils/utilities";

import "./AddMessage.styles.scss";

export default class AddMessage extends React.Component {
  state = {
    message: "",
    user: ""
  };

  componentDidMount = async () => {
    const { displayName, uid, email } = this.props;
    this.setState({
      user: {
        displayName,
        uid,
        email
      }
    });
  };
  handleSubmit = async e => {
    e.preventDefault();
    if (!this.state.message) {
      alert("Please enter a message!");
    } else {
      const { message } = this.state;
      const { displayName, uid, email } = this.props;
      let post = {
        message,
        user: {
          displayName,
          uid,
          email
        },
        createdAt: new Date()
      };
      firestore.collection("messages").add(post);

      this.setState({ message: "" });
    }
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className="AddPost">
        <span style={{ color: "green" }}>â†’</span>
        {this.state.message.length < 1 ? (
          <div className="blink" style={{ opacity: 1 }}></div>
        ) : (
          <div className="blink" style={{ opacity: 0 }}></div>
        )}
        <input
          style={styleFunc(this.state.message)}
          className="add-message"
          type="text"
          autocomplete="off"
          name="message"
          value={inputHelper(this.state.message)}
          onChange={e => this.setState({ message: e.target.value })}
        />
      </form>
    );
  }
}
