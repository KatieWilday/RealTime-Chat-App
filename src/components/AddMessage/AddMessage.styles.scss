$button-color: #1ab188;
$button-dark: darken($button-color, 5%);
$window-black: #30353a;

.AddPost {
  margin-top: 15px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
}

.Message {
  margin-bottom: 25px;
}

.add-message {
  position: relative;
  bottom: 0.5px;
  left: 25px;
  top: -22.5px;
  caret-color: transparent !important;
  height: 10px;
  width: 600px;
  font-size: 1.05em;
  background: $window-black;
  border: none;
  color: #fff;
  opacity: 1;

  &:focus {
    outline: none;
  }
}

.blink {
  position: absolute;
  left: 25px;
  color: #fff;
  z-index: 500;

  &::after {
    content: "|";
    -webkit-animation: blink 2s steps(1) infinite;
    animation: blink 2s steps(1) infinite;
  }
}

@-webkit-keyframes blink {
  0% {
    color: #fff;
  }
  50% {
    color: transparent;
  }
  100% {
    color: #fff;
  }
}

@keyframes blink {
  0% {
    color: #fff;
  }
  50% {
    color: transparent;
  }
  100% {
    color: #fff;
  }
}
