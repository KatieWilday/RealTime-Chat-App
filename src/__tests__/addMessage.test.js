import React from "react";
import AddMessage from "../components/AddMessage/AddMessage.component";
import { create } from "react-test-renderer";

test("snapshot", () => {
  const tree = create(<AddMessage />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("creates post", () => {
  const tree = create(<AddMessage search={() => {}} />);
  const instance = tree.getInstance();

  expect(instance.state.message).toBe("");
  instance.handleSubmit("test");
  expect(instance.state.message).toBe("");
});
