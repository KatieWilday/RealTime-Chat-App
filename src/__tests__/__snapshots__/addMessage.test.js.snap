// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<form
  className="AddPost"
  onSubmit={[Function]}
>
  <span
    style={
      Object {
        "color": "green",
      }
    }
  >
    â†’
  </span>
  <div
    className="blink"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <input
    autoComplete="off"
    className="add-message"
    name="message"
    onChange={[Function]}
    style={
      Object {
        "height": 30,
      }
    }
    type="text"
    value=""
  />
</form>
`;
